app/models/profile.rb

```diff
@@ -1,4 +1,12 @@
 class Profile < ApplicationRecord
   belongs_to :user
   has_many :ingestions
+  attr_reader :BMR
+  before_save :calculate_bmr
```

Comment:
Styling
Separate associations, attributes and callbacks with empty line. It will be easier to read.